core/src/main/java/io/confluent/kafka/schemaregistry/id/IdGenerator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry.id;
+
+import io.confluent.kafka.schemaregistry.client.rest.entities.Schema;
+import io.confluent.kafka.schemaregistry.exceptions.IdGenerationException;
+import io.confluent.kafka.schemaregistry.rest.SchemaRegistryConfig;
+import io.confluent.kafka.schemaregistry.storage.SchemaKey;
+import io.confluent.kafka.schemaregistry.storage.SchemaValue;
+
+/**
+ * Internal interface used for generating id while registering schema.
+ */
+public interface IdGenerator {
+
+  /**
+   * Provides the id for the schema.
+   *
+   * @param schema the schema being registered; never {@code null}
+   * @return the identifier for the schema; never {@code null}
+   * @throws IdGenerationException exception when id can't be generated
+   */
+  int id(Schema schema) throws IdGenerationException;
+
+  /**
+   * Configure the underlying generator.
+   *
+   * @param config Schema Registry Configs
+   */
+  void configure(SchemaRegistryConfig config);
+
+  /**
+   * Initialize the underlying generator.
+   *
+   * @throws IdGenerationException exception when IdGenerator can't be initialized
+   */
+  void init() throws IdGenerationException;
```

Comment:
I assume you are referring to the init() method here. Blck indefinitely would be a characteristic of the implementation but since this is init() the implementations would need to block. Also, I don't see a reason to throw interrupted exception in this scenario.